demoapps/passwordlogin/java/org/openyolo/demoapp/passwordlogin/MainViewModel.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 The OpenYOLO Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openyolo.demoapp.passwordlogin;
+
+import android.app.Application;
+import android.databinding.ObservableField;
+import android.support.annotation.AnyThread;
+import android.support.annotation.MainThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.WorkerThread;
+import android.view.View;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.openyolo.demoapp.passwordlogin.userdata.User;
+import org.openyolo.demoapp.passwordlogin.userdata.UserDataSource;
+
+/**
+ * The view model for the main activity, which is displayed when the user has authenticated
+ * successfully.
+ */
+@WorkerThread
+public final class MainViewModel extends ObservableViewModel {
+
+    /**
+     * The user's display name for display on screen, if known.
+     */
+    public final ObservableField<String> displayName = new ObservableField<>("");
+
+    /**
+     * The user's email address for display on screen.
+     */
+    public final ObservableField<String> email = new ObservableField<>("");
+
+    /**
+     * The user's display picture for display on screen, if known.
+     */
+    public final ObservableField<String> displayPicture = new ObservableField<>("");
+
+    private final AtomicBoolean mFirstLoad = new AtomicBoolean(true);
+    private final UserDataSource mUserDataSource;
+    private MainNavigator mNavigator;
+
+    /**
+     * Creates the view model, with the required application reference.
+     */
+    public MainViewModel(@NonNull Application application) {
+        super(application);
+        mUserDataSource = ((OpenYoloDemoApplication)application).getUserRepository();
+    }
+
+    /**
+     * Specifies the navigator for use in communicating with the activity environment.
+     */
+    @MainThread
+    public void setNavigator(@NonNull MainNavigator navigator) {
```

Comment:
The documentation (https://developer.android.com/reference/android/arch/lifecycle/ViewModel.html) I found suggests a ViewModel should *"scary bold text"* never have a reference it's associated activity.  It looks like that's what you're doing here.  IMO it seems nice to have the business logic of transitions in here.  Any idea why they suggest avoiding this?